### PR TITLE
Build fixes/enhancements

### PR DESCRIPTION
**Pull request by juan-lunarg**
Copying lots of build improvements over from VVL

EX. Now its possible to use Ninja on Windows with the fixes to update_deps.py

Also the performance enhancements to scripts/CMakeLists.txt

### CODE REVIEW

File: scripts/known_good.json [hunk 1]
```diff
@@ -28,19 +28,14 @@
             "sub_dir": "Vulkan-ValidationLayers",
             "build_dir": "Vulkan-ValidationLayers/build",
             "install_dir": "Vulkan-ValidationLayers/build/install",
-            "commit": "main",
```
Comment (juan-lunarg) [inline scripts/known_good.json:31]: Note. This little bug took me a bit to figure out.

**Comment by juan-lunarg** [comment]
Adding ccache to unix builds and using Ninja on Windows should help with GHA performance a fair amount

**Comment by juan-lunarg** [comment]
> CI VulkanProfiles build # 2524 failed.

Looks like a CI issue

File: scripts/update_deps.py [hunk 2]
```diff
@@ -656,7 +645,7 @@ def main():
         '--platform',
         dest='platform',
         type=str.lower,
-        help='Select build platform to use; generally only useful for "android"',
```
Comment (juan-lunarg) [inline scripts/update_deps.py:659]: Needed for Android jenkins build.

**Review by christophe-lunarg — APPROVED** [review]

**Pull request merged**